src/components/utils/include/utils/logger/logger_impl.h

```diff
@@ -0,0 +1,120 @@
+#pragma once
+#include "utils/ilogger.h"
+
+#include <assert.h>
+#include <functional>
+#include <iostream>
+
+#include "utils/logger_status.h"
+#include "utils/threads/message_loop_thread.h"
+
+namespace logger {
+
+typedef std::queue<SDLLogMessage> LogMessageQueue;
+
+typedef threads::MessageLoopThread<LogMessageQueue>
+    LogMessageLoopThreadTemplate;
+
+template <typename T>
+using LoggerPtr = std::unique_ptr<T, std::function<void(T*)> >;
```

Comment:
LoggerPtr is bad name, I do not understand what actually is doing

Reply:
Renamed